src/components/ImportRecord/ImportRecord.js

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage, injectIntl } from 'react-intl';
+import ky from 'ky';
+import { stripesConnect, withOkapiKy, CalloutContext } from '@folio/stripes/core';
+import { LoadingView } from '@folio/stripes/components';
+
+
+class ImportRecord extends React.Component {
+  static propTypes = {
+    id: PropTypes.string,
+    stripes: PropTypes.shape({
+      logger: PropTypes.object.isRequired,
+    }).isRequired,
+    resources: PropTypes.shape({
+      query: PropTypes.shape({
+        xid: PropTypes.string,
+      }).isRequired,
+    }).isRequired,
+    mutator: PropTypes.shape({
+      query: PropTypes.shape({
+        update: PropTypes.func.isRequired,
+      }).isRequired,
+    }).isRequired,
+    okapiKy: PropTypes.func.isRequired,
+    intl: PropTypes.shape({
+      formatMessage: PropTypes.func.isRequired,
+    }).isRequired,
+  };
+
+  static contextType = CalloutContext;
+
+  static manifest = Object.freeze({
+    query: {},
+  });
+
+  constructor(props) {
+    super(props);
+    this.loading = false;
+  }
+
+  componentDidUpdate() {
+    if (!this.loading) {
+      const xid = this.props.resources.query?.xid;
+      this.loading = true;
+      this.loadExternalRecord(xid);
+    }
+  }
+
+  loadExternalRecord = (xid) => {
+    const { id, intl } = this.props;
+
+    this.props.okapiKy('copycat/imports', {
+      timeout: 30000,
+      method: 'POST',
+      json: {
+        externalIdentifier: xid,
+        internalIdentifier: id,
+        profileId: 'f26df83c-aa25-40b6-876e-96852c3d4fd4' // XXX hardwired to UUID in mod-copycat reference data
+      },
+    }).then(res => {
+      // No need to check res.ok, as ky throws non-2xx responses
+      res.json().then(json => {
```

Comment:
Could you extract that to a new function to avoid using nested `then`? Another approach here could be to mark `loadExternalRecord` as `async` and use `await` for async calls.

Reply:
Addressed in f3bf18ede334c20199f3bd802f06eed89c3f86f0

Reply:
And also in 478934563a2fbd264460881d9af06e9901082632